### PR TITLE
List of activities register

### PR DESCRIPTION
**Pull request by RoxaneBurri**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -71,7 +71,7 @@ dependencies {
     implementation 'com.google.android.gms:play-services-location:18.0.0'
     implementation "androidx.fragment:fragment:1.3.2"
     implementation 'androidx.room:room-runtime:2.2.6'
-    implementation 'com.google.firebase:firebase-database:18.0.1'
+    implementation 'com.google.firebase:firebase-database:19.5.1'
```
Comment (Walfar) [inline app/build.gradle:75]: Changing the version worked for me, so I'm really surprised you still have the Firebase SDK version error from Cirrus... I also tried using the latest version (i.e 19.7.0), but it still wasn't working. You could temporarly try to comment the test blocking the run with this error, to see if this would do the trick.

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java [hunk 2]
```diff
@@ -159,12 +158,17 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                 FirebaseInteraction.logoutIfUserNonNull(fAuth, this);
                 break;
             case R.id.nav_start_activity:
+                Navigation.startActivity(this.navigationView);
```
Comment (jlstoupy98) [inline app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java]: Why did you add this line?

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java [hunk 3]
```diff
@@ -159,12 +158,17 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
...
                 break;
             case R.id.nav_chat:
                 Navigation.goToChat(this.navigationView);
                 break;
+            case R.id.nav_list_activities:
+                Navigation.goToListOfActivities(this.navigationView);
```
Comment (jlstoupy98) [inline app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:173]: You need to add a "finish()" after this navigation to avoid overlap of activities. I already add this to all other items in my PR

File: app/build.gradle (hunk 1 above)
Comment (RoxaneBurri) [reply app/build.gradle:75]: Yes, and my first commit works, so I don't understand why it's fail now. Last week  with Jean-Luc and Anthony, we already try to comment the test, but an other test in the chat produce this error.

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                FirebaseInteraction.logoutIfUserNonNull(fAuth, this);
+                break;
+            case R.id.nav_start_activity:
+                Navigation.startActivity(this.navigationView);
+                break;
+            case R.id.nav_chat:
+                Navigation.goToChat(this.navigationView);
+                break;
+            case R.id.nav_list_activities:
+                Navigation.goToListOfActivities(this.navigationView);
```
Comment (jlstoupy98) [inline app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java]: Is it really useful to make a redirection to this activities while we already are on the ListActivity ?

File: app/src/main/java/com/sdp/movemeet/MainActivity.java [hunk 5]
```diff
@@ -88,9 +76,9 @@ public void createDrawer(){
     public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
         switch (menuItem.getItemId()) {
             case R.id.nav_home:
+                Navigation.goToHome(this.navigationView);
```
Comment (jlstoupy98) [inline app/src/main/java/com/sdp/movemeet/MainActivity.java]: Same as before, I think we don't need to make a redirection when we are already on the activity

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class ListActivityTest {
+
+    private FirebaseAuth fAuth;
+    private FirebaseDatabase database;
+
+    @Test
+    public void listOfActivityTest() {
```
Comment (jlstoupy98) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java:28]: What do you test with this test ?

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java (hunk 2 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java]: I think it's because of the code style update

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java (hunk 3 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:173]: Okay !

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java (hunk 4 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java]: You right, I remove it

File: app/build.gradle [hunk 7]
```diff
@@ -71,7 +71,7 @@ dependencies {
     implementation 'com.google.android.gms:play-services-location:18.0.0'
     implementation "androidx.fragment:fragment:1.3.2"
     implementation 'androidx.room:room-runtime:2.2.6'
-    implementation 'com.google.firebase:firebase-database:18.0.1'
+    implementation 'com.google.firebase:firebase-database:19.7.0'
```
Comment (Antho1426) [inline app/build.gradle]: Nice try again 😂

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java (hunk 6 above)
Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java:28]: I am also interested to know!

Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java:28]: I just want to launch the class, because the page is empty for the moment

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+                break;
+        }
+        drawerLayout.closeDrawer(GravityCompat.START);
+        return true;
+    }
+
+    public void handleRegisterUser() {
+        fAuth = FirebaseAuth.getInstance();
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java:104]: It would be nice if we could refactor this function appearing in a lot of our classes in the future. I will try to think about it!

File: app/src/main/java/com/sdp/movemeet/Navigation/Navigation.java [hunk 9]
```diff
@@ -45,4 +46,10 @@ public static void goToChat(View view) {
         mContext.startActivity(intent);
     }
 
+    public static void goToListOfActivities(View view) {
+        Context mContext = view.getContext();
+        Intent intent = new Intent(mContext, ListActivity.class);
+        mContext.startActivity(intent);
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/Navigation/Navigation.java]: Didn't we say on Tuesday that our agreed convention was **NOT** to use the "m" prefix?

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java (hunk 4 above)
Comment (bitshrine) [reply app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java]: Also, @jlstoupy98, should this Navigation function include the `finish()` you mentioned for the `ActivityDescriptionActivity`?

File: app/src/main/java/com/sdp/movemeet/MainActivity.java (hunk 5 above)
Comment (bitshrine) [reply app/src/main/java/com/sdp/movemeet/MainActivity.java]: Same comment regarding adding the `finish()`

**Review by Antho1426 — APPROVED** [review]

File: app/src/main/java/com/sdp/movemeet/ProfileActivity.java [hunk 10]
```diff
@@ -131,8 +131,12 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
             case R.id.nav_chat:
                 Navigation.goToChat(this.navigationView);
                 break;
+            case R.id.nav_list_activities:
+                Navigation.goToListOfActivities(this.navigationView);
```
Comment (bitshrine) [inline app/src/main/java/com/sdp/movemeet/ProfileActivity.java:141]: + `finish()` (I hope these are actually required, or I'm just marking them for nothing)

File: app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java [hunk 11]
```diff
@@ -176,8 +168,12 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
             case R.id.nav_chat:
                 Navigation.goToChat(this.navigationView);
                 break;
+            case R.id.nav_list_activities:
+                Navigation.goToListOfActivities(this.navigationView);
+                break;
```
Comment (bitshrine) [inline app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java:180]: ... and another `finish()`

File: app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java [hunk 12]
```diff
@@ -361,7 +358,7 @@ private Activity validateActivity() {
 
         String organizerId = FirebaseAuth.getInstance().getCurrentUser().getUid();
         Activity activity = new Activity(
-                organizerId + " || " +  date, organizerId, title, nParticipants,
+                organizerId + " || " + date, organizerId, title, nParticipants,
```
Comment (bitshrine) [inline app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java:368]: We should change this way of defining an activity ID. Perhaps we should have a hashing function which hashes both the organizer, sport and creation date.

File: app/src/main/java/com/sdp/movemeet/chat/ChatActivity.java [hunk 13]
```diff
@@ -235,6 +234,10 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                 Navigation.startActivity(this.navigationView);
                 break;
             case R.id.nav_chat:
+                Navigation.goToChat(this.navigationView);
+                break;
+            case R.id.nav_list_activities:
+                Navigation.goToListOfActivities(this.navigationView);
```
Comment (bitshrine) [inline app/src/main/java/com/sdp/movemeet/chat/ChatActivity.java:246]: Another Navigation instance :)

**Review by bitshrine — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java (hunk 6 above)
Comment (Walfar) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java:28]: I'm not sure to understand the purpose of this test, what should it be used for ?

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+        fullName = hView.findViewById(R.id.text_view_profile_name);
+        phone = hView.findViewById(R.id.text_view_profile_phone);
+        email = hView.findViewById(R.id.text_view_profile_email);
+
+        toggle.syncState();
+        navigationView.setNavigationItemSelectedListener(this::onNavigationItemSelected);
+        navigationView.setCheckedItem(R.id.nav_home);
+
+    }
+
```
Comment (Walfar) [inline app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java:68]: I'll try refactoring those methods used in many classes ;)

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+import androidx.drawerlayout.widget.DrawerLayout;
+
+import com.google.android.material.navigation.NavigationView;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.sdp.movemeet.Backend.FirebaseInteraction;
+import com.sdp.movemeet.Navigation.Navigation;
+import com.sdp.movemeet.R;
+
+public class ListActivity extends AppCompatActivity {
```
Comment (Walfar) [inline app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java:25]: What is this activity used for ?

File: app/src/main/java/com/sdp/movemeet/Navigation/Navigation.java [hunk 16]
```diff
@@ -45,4 +46,10 @@ public static void goToChat(View view) {
         mContext.startActivity(intent);
     }
 
+    public static void goToListOfActivities(View view) {
+        Context mContext = view.getContext();
+        Intent intent = new Intent(mContext, ListActivity.class);
```
Comment (Walfar) [inline app/src/main/java/com/sdp/movemeet/Navigation/Navigation.java]: You can directly put the intent in the startActivity call

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java (hunk 4 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java]: I add the finish in the last commit

**Review by Antho1426 — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java (hunk 6 above)
Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java:28]: I will update the test when the activity show something, because for now the activity do nothing (I only create a page, and add it in the menu)

File: app/build.gradle (hunk 7 above)
Comment (RoxaneBurri) [reply app/build.gradle]: I still hope

File: app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java (hunk 12 above)
Comment (Walfar) [reply app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java:368]: yep, good idea. And maybe we can call it directly inside the Activity constructor

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java (hunk 8 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java:104]: Yes ! This is a lot of repetitions, Victor have may be an idea how to do it

File: app/src/main/java/com/sdp/movemeet/Navigation/Navigation.java (hunk 9 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Navigation/Navigation.java]: My bad, I am sorry

File: app/src/main/java/com/sdp/movemeet/ProfileActivity.java (hunk 10 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/ProfileActivity.java:141]: Done in last commit

File: app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java (hunk 11 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/UploadActivityActivity.java:180]: also done

File: app/src/main/java/com/sdp/movemeet/chat/ChatActivity.java (hunk 13 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/chat/ChatActivity.java:246]: done too

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java (hunk 6 above)
Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ListActivityTest.java:28]: The class in empty, so the test just launch the class, but I will update this test during my next task

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java (hunk 14 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java:68]: Nice !

File: app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java (hunk 15 above)
Comment (RoxaneBurri) [reply app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java:25]: I add a comment in my last commit, but I can explain more
